File: python/meteroid/models/invoice_event_data.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .customer_id import CustomerId
from .invoice_id import InvoiceId
from .invoice_status import InvoiceStatus


@dataclasses.dataclass
class InvoiceEventData(BaseModel):
    created_at: datetime

    currency: str

    custom_properties: t.Dict[str, t.Any]
    """User-defined custom property values, keyed by definition key."""

    customer_id: CustomerId

    invoice_id: InvoiceId

    status: InvoiceStatus

    tax_amount: int

    total: int

    consolidated_into_invoice_id: t.Optional[InvoiceId] = None

    invoice_number: t.Optional[str] = None
    """Absent while the invoice is a draft — the number is assigned at finalization."""

    parent_invoice_id: t.Optional[InvoiceId] = None

File: go/invoice_event_data.go
// this file is @generated
package meteroid

import (
	"encoding/json"
	"time"
)

type InvoiceEventData struct {
	ConsolidatedIntoInvoiceId *InvoiceId `json:"consolidated_into_invoice_id,omitempty"`

	// RFC 3339 timestamp.
	CreatedAt time.Time `json:"created_at"`

	Currency string `json:"currency"`

	// User-defined custom property values, keyed by definition key.
	CustomProperties json.RawMessage `json:"custom_properties"`

	CustomerId CustomerId `json:"customer_id"`

	InvoiceId InvoiceId `json:"invoice_id"`

	// Absent while the invoice is a draft — the number is assigned at finalization.
	InvoiceNumber *string `json:"invoice_number,omitempty"`

	ParentInvoiceId *InvoiceId `json:"parent_invoice_id,omitempty"`

	Status InvoiceStatus `json:"status"`

	TaxAmount int64 `json:"tax_amount"`

	Total int64 `json:"total"`
}
